### PR TITLE
`timeseries2velocity`: add `--polyline` option

### PR DESCRIPTION
**Pull request by yunjunz**
**Description of proposed changes**

This PR add the `timeseries2velocity.py --polyline` option (and for `tsview.py` as well). Given one or multiple dates, this option will estimate an additional linear slope starting from the specified date, to catch the linear velocity change of a time-series. The result is saved in the velocity.h5 file as a dataset named `velocityPost{YYYYMMDD}`, which is the linear velocity during this segment time period, i.e. the sum of the velocity for the entire period (from the regular polynomial fitting), plus all the additional velocity change before & during the current segment periods.

Detailed changes are as below:

+ `utils.time_func`: construct a design matrix for polyline fitting with the manually specified starting date for each additional line segment

+ `utils.arg_utils.add_timefunc_argument()`: add `--polyline` option

+ `timeseries2velocity`:
   - `model2hdf5_dataset()`: support the polyline function and save the total/cumulative velocity for each polyline segment for easy visualization and interpretation.
   - check input `--polynomial` and `--polyline` options

+ `cli.timeseries2velocity`: add example usage

@yuankailiu I would appreciate it if you could take a look at these changes above when you got a chance?

Other changes not relevant to `--polyline`:

+ `utils.ptime`:
   - `date_list2tbase()`: add `ref_date` arg
   - `date_list2vector()`: add `seconds` arg

+ `utils.arg_utils.add_figure_argument()`: add `--cbar-ticks` option for accurate control on the colorbar tick positions.

+ `simulation.iono.lalo_ground2iono()`: more comments

+ `tsview`: printout msg minor adjustment
   - remove redundant print cmd. Now the tsview cmd msg will always be printed out.
   - always printout the save fig/txt file info, as it's very informative

+ `objects.gps`: change the default folder to store the site list file from `./` to `./GPS`.

**Reminders**

- [x] Pass Pre-commit check (green)
- [x] Pass Codacy code review (green)
- [x] Pass [Circle CI test](https://app.circleci.com/pipelines/github/yunjunz/MintPy/2091/workflows/7cc1e074-2b09-4cf2-a5ff-ddddded9faa7/jobs/2096) (green)
- [x] Make sure that your code follows our style. Use the other functions/files as a basis.
- [x] If modifying functionality, describe changes to function behavior and arguments in a comment below the function declaration.
- [x] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Comment by yuankailiu** [comment]
@yunjunz, thanks for the PR. Yes I will take a look.

File: src/mintpy/utils/time_func.py [hunk 1]
```diff
@@ -111,6 +114,12 @@ def inps2model(inps, date_list=None, print_msg=True):
             if not (ymin < ptime.yyyymmdd2years(d_step) < ymax):
                 raise ValueError(f'input step date ({d_step}) exceeds date limit: ({dmin} / {dmax})!')
 
+    if inps.polyline:
+        # check 1 - min/max limit
+        for d_start in inps.polyline:
+            if not (ymin < ptime.yyyymmdd2years(d_start) < ymax):
+                raise ValueError(f'input step date ({d_start}) exceeds date limit: ({dmin} / {dmax})!')
```
Comment (yuankailiu) [inline src/mintpy/utils/time_func.py:121]: input "polyline" date here?

**Review by yuankailiu — APPROVED** [review]
Thank you. I think it looks good to me.

File: src/mintpy/utils/time_func.py (hunk 1 above)
Comment (yuankailiu) [reply src/mintpy/utils/time_func.py:121]: I only spot this printed message typo. Otherwise, you can go ahead to merge it!

Comment (yunjunz) [reply src/mintpy/utils/time_func.py]: Fixed. Thanks, kai.

**Pull request merged**